flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/ConnectTableDescriptor.java

```diff
@@ -61,36 +103,77 @@ public ConnectTableDescriptor withSchema(Schema schema) {
 	 * a table under the given name.
 	 *
 	 * @param name table name to be registered in the table environment
+	 * @deprecated use {@link #createTemporaryTable(String)}
 	 */
+	@Deprecated
```

Comment:
maybe mention the difference, otherwise I see questions asking why there is no distinction between source and sink anymore. we can mention in `createTemporaryTable` that the search for a table factory is lazy now?